web/public/js/controllers/dojos-map-controller.js

```diff
@@ -198,22 +222,40 @@ function cdDojosMapCtrl($scope, $window, $state, $stateParams, $translate, $geol
   }
 
   function addMarkersToMap(dojos) {
-    if (markerClusterer) markerClusterer.clearMarkers();
-    _.each(dojos, function (dojo) {
-      if (dojo.geoPoint || dojo.geo_point) {
-        var pinColor = dojo.private === 1 ? 'FF0000' : '008000';
-        var marker = new google.maps.Marker({
-          map: $scope.model.map,
-          dojoName: dojo.name,
-          dojoId: dojo.id,
-          icon: 'img/marker' + pinColor +'.png',
-          position: new google.maps.LatLng(dojo.geoPoint && dojo.geoPoint.lat || dojo.geo_point.lat, dojo.geoPoint && dojo.geoPoint.lon || dojo.geo_point.lon)
+    return new Promise(function(resolve, reject) {
+      if (markerClusterer) markerClusterer.clearMarkers();
+      _.each(dojos, function (dojo) {
+        if (dojo.geoPoint || dojo.geo_point) {
+          var pinColor = dojo.private === 1 ? 'FF0000' : '008000';
+          var marker = new google.maps.Marker({
+            map: $scope.model.map,
+            dojoName: dojo.name,
+            dojoId: dojo.id,
+            icon: 'img/marker' + pinColor +'.png',
+            position: new google.maps.LatLng(dojo.geoPoint && dojo.geoPoint.lat || dojo.geo_point.lat, dojo.geoPoint && dojo.geoPoint.lon || dojo.geo_point.lon)
+          });
+          marker.dojo = dojo;
+          $scope.markers.push(marker);
+        }
+      });
+      markerClusterer = new MarkerClusterer($scope.model.map, $scope.markers, {imagePath:'components/google-maps-utility-library-v3/markerclusterer/images/m'});
```

Comment:
It will. It returns the Promise, which when chained with other promises will then get executed.